oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1255,28 +1305,40 @@ public void removeGarbage(final VersionGCStats stats) {
                 }
                 if (!isDetailedGCDryRun) {
                     // only delete these in case it is not a dryRun
-                    List<NodeDocument> oldDocs = ds.findAndUpdate(NODES, updateOpList);
-                    int deletedProps = oldDocs.stream().filter(Objects::nonNull).mapToInt(d -> deletedPropsCountMap.getOrDefault(d.getId(), 0)).sum();
-                    int updatedDocs = (int) oldDocs.stream().filter(Objects::nonNull).count();
-                    stats.updatedDetailedGCDocsCount += updatedDocs;
-                    stats.deletedPropsCount += deletedProps;
-                    stats.deletedUnmergedBCCount += deletedUnmergedBCSet.size();
-
-                    if (log.isDebugEnabled()) {
-                        log.debug("Updated [{}] documents, deleted [{}] properties, deleted [{}] unmergedBranchCommits",
-                                updatedDocs, deletedProps, deletedUnmergedBCSet.size());
+
+                    if (!orphanOrDeletedRemovalList.isEmpty()) {
+                        ds.remove(NODES, orphanOrDeletedRemovalList);
+                        final int removedSize = orphanOrDeletedRemovalList.size();
+                        stats.updatedDetailedGCDocsCount += removedSize;
+                        stats.deletedDocGCCount += removedSize;
```

Comment:
Please add debug logging about the orphan nodes deletions.

Reply:
done also in https://github.com/apache/jackrabbit-oak/pull/1315/commits/74ef58d446f770bdfc2723b0039409ca963c3c4d